### ISSUE TITLE
Outdated vulnerability information for vault CVEs

### ISSUE THREAD

**Issue by samj1912** [anchore/grype#609]
**What happened**: 
```
grype ghcr.io/kyverno/kyverno:1.6-dev-latest
 ✔ Vulnerability DB        [no update available]
 ✔ Parsed image            
 ✔ Cataloged packages      [221 packages]
 ✔ Scanned image           [23 vulnerabilities]

NAME                            INSTALLED  FIXED-IN  VULNERABILITY   SEVERITY 
github.com/hashicorp/vault/api  v1.3.1               CVE-2020-10660  Medium    
github.com/hashicorp/vault/api  v1.3.1               CVE-2020-10661  Critical  
github.com/hashicorp/vault/api  v1.3.1               CVE-2020-13223  High      
github.com/hashicorp/vault/api  v1.3.1               CVE-2020-16250  Critical  
github.com/hashicorp/vault/api  v1.3.1               CVE-2020-16251  Critical  
github.com/hashicorp/vault/api  v1.3.1               CVE-2020-25594  Medium    .....
```
**What you expected to happen**:

Expected grype to say that `CVE-2020-10661` is fixed in 1.3.4 as per https://nvd.nist.gov/vuln/detail/CVE-2020-10661

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

```
grype version
Application:          grype
Version:              0.32.0
Syft Version:         v0.36.0
BuildDate:            2022-01-20T18:48:06Z
GitCommit:            3ba7e56e42fddca0fd944986596068e429d448fa
GitTreeState:         clean
Platform:             darwin/amd64
GoVersion:            go1.16.13
Compiler:             gc
Supported DB Schema:  3
```

**Comment by westonsteimel** [comment]
Nope, I updated my database locally so that `version_constraint`=`>= 0.11.0, <= 1.3.4 || >= 0.11.0, <= 1.3.4` and ended up with the same result

**Comment by westonsteimel** [comment]
Ah, I think maybe it's because we have a v in the version when generating CPE to match against!

```json
{
   "vulnerability": {
    "id": "CVE-2020-10661",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2020-10661",
    "namespace": "nvd",
    "severity": "Critical",
    "urls": [
     "https://www.hashicorp.com/blog/category/vault/",
     "https://github.com/hashicorp/vault/blob/master/CHANGELOG.md#134-march-19th-2020"
    ],
    "description": "HashiCorp Vault and Vault Enterprise versions 0.11.0 through 1.3.3 may, under certain circumstances, have existing nested-path policies grant access to Namespaces created after-the-fact. Fixed in 1.3.4.",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:M/Au:N/C:P/I:P/A:N",
      "metrics": {
       "baseScore": 5.8,
       "exploitabilityScore": 8.6,
       "impactScore": 4.9
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:N",
      "metrics": {
       "baseScore": 9.1,
       "exploitabilityScore": 3.9,
       "impactScore": 5.2
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "matcher": "stock-matcher",
  "searchedBy": {
      "namespace": "nvd",
      "cpes": [
       "cpe:2.3:a:hashicorp:vault:v1.3.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "versionConstraint": ">= 0.11.0, <= 1.3.3 || >= 0.11.0, <= 1.3.3 (unknown)",
      "cpes": [
       "cpe:2.3:a:hashicorp:vault:*:*:*:*:-:*:*:*",
       "cpe:2.3:a:hashicorp:vault:*:*:*:*:enterprise:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "github.com/hashicorp/vault/api",
    "version": "v1.3.1",
    "type": "go-module",
    "locations": [
     {
      "path": "/kyverno",
      "layerID": "sha256:87ea2318e02c196559747cdfb16c27132cf54eaa06241faf1a025f656f25247e"
     }
    ],
    "language": "go",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:hashicorp:vault:v1.3.1:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:golang/github.com/hashicorp/vault/api@v1.3.1",
    "metadata": null
   }
  }
```

I suspect that if it was `"cpe:2.3:a:hashicorp:vault:1.3.1:*:*:*:*:*:*:*"` rather than `"cpe:2.3:a:hashicorp:vault:v1.3.1:*:*:*:*:*:*:*"` that it might work as expected?

**Comment by spiffcs** [comment]
@samj1912 I'm going to close this now since both the vault vulnerabilities have been removed from syft (and by relation grype) and we've updated our CPE logic so that we're no longer namespacing the vault vulnerabilities incorrectly.

If you have questions, comments or think this has been closed in error just let me know!
https://github.com/anchore/syft/pull/1003 <-- Removal of vault vulnerabilities
https://github.com/anchore/syft/pull/1006 <-- Longer CPE to avoid go submodules included in parent CPE
